### DISCUSSION TITLE
Typed property App\Http\Livewire\SettingsCheckin::$auto_checkout must be int, string used

### DISCUSSION THREAD

**Discussion by torian257x** [livewire/livewire#3579] [7. Bugs (archived)] [answered]
Version

>"name": "livewire/livewire",
>"version": "v2.5.5",

I get

>  Typed property App\Http\Livewire\SettingsCheckin::$auto_checkout must be int, string used 

Problem: I can only reproduce it ONCE. I saw it on the server, went to local development, and got the same issue. 

I then added a 

```
    protected $casts = [
        'auto_checkout' => 'int',
    ]
```

tried again, now no error, YAY! but wait... there are other ints on THE SAME component, that just work fine without cast.

Then I removed the cast> no more error. So now I don't get the error at all, even if the cast is gone (like it was before).

Looking at the page itself

component gets rendered with an `@if` in case checkout is required

```
               <input type="checkbox" class="custom-control-input" name="is_checkout" value="1" wire:model="is_checkout" @if($is_checkout == 1) checked @endif>

                    @if ($is_checkout == 1)
                        <div class="">
                            <input type="number" class="form-control" name="data[auto_checkout]" placeholder="2"
                                   id="auto_checkout" value="{{$auto_checkout}}"
                                   min="1" max="12" wire:model.debounce.500ms="auto_checkout">
                        </div>
                    @endif
```

is there maybe a caching issue? e.g. on the very first call, it does cause an issue, but every other call works fine?

How can I clear my cache?

**Comment by joshhanley** [comment, marked as answer]
@torian257x not sure why adding the casts would have helped, as they were removed from Livewire in V2 https://laravel-livewire.com/docs/2.x/upgrading#casters

If there is any caching going on it will be the compile blade views, you could try doing `php artisan view:clear`.

What you will find is probably the cause, is the fact that html inputs return strings, even if they are empty it will return an empty string. So if you have an `int` type hint on a property, Livewire will try and fill the property with what it gets from the form (an empty string), which will then break the `int` type hint, hence the error.

The only way I've found to work around it, if you want to keep the type hint, is to do a check on the property in an updating hook and if it's set to an empty string, then convert it to null.

So if we have `$foo` type hinted to an int, then in the `updatingFoo` hook, we can accept the incoming value by reference `&$value` and update it to null if it's an empty string.

```php
public ?int $foo;

public function updatingFoo(&$value) {
  if ($value == "") {
    $value = null;
  }
}
```

Hope this helps!

**Reply by torian257x** [reply]
the casts were removed? that's saddening. Especially since the explanation is a bit "yer a bunch of donkeys"

but thanks a lot for the pointer about null, it was indeed empty value that was a problem, not string per se. Highly appreciate your feedback.

That's why I couldn't reproduce, because it cast it perfectly fine in the other cases and no error, I just couldn't reproduce myself removing all values by accident.

So basically, don't type hint? a bit a shame... that's why all docs lack the type hinting I guess
